### ISSUE TITLE
Improve documentation on project execution loop

### ISSUE THREAD

**Issue by csadorf** [glotzerlab/signac-docs#18]
**[Original report](https://bitbucket.org/glotzer/signac-flow/issue/32) by Carl Simon Adorf (Bitbucket: [csadorf](https://bitbucket.org/csadorf), GitHub: [csadorf](https://github.com/csadorf)).**

----------------------------------------

The execution loop that flow enters when calling `$ python project.py run` and all of its variants is not well documented leading to confusion among users.

**Comment by syjlee** [comment]
@csadorf *Original report* above is not available since this issue was created before the repositories moved to Github. Were there important details I need to keep eyes on in there?

**Comment by csadorf** [comment]
> @csadorf _Original report_ above is not available since this issue was created before the repositories moved to Github. Were there important details I need to keep eyes on in there?

No, there was not more information available. However, I am going to write up a brief summary of what I believe we can do to address this issue.

**Comment by csadorf** [comment]
I think we should add another section *Run operations* after the [The Project Status ](http://docs.signac.io/en/latest/flow-project.html#the-project-status) section that explains exactly how the execution loop works. The current explanation is somewhat scattered over different places.

I would start by reading and understanding the [`FlowProject.run()`](https://github.com/glotzerlab/signac-flow/blob/develop/flow/project.py#L1436) implementation and would then try to translate that concisely into plain english. We should consider to also update the doc-string of that function as well.

**Comment by csadorf** [comment]
@syjlee Were you able to make any progress on this?
